### PR TITLE
Read only tikv

### PR DESCRIPTION
**Pull request by vangheem**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: nucliadb/nucliadb/ingest/settings.py [hunk 1]
```diff
@@ -63,6 +63,10 @@ class DriverSettings(BaseSettings):
         default=20,
         description="PostgreSQL max pool size. The maximum number of connections to the PostgreSQL server.",
     )
+    driver_tikv_connection_pool_size: int = Field(
+        default=3,
+        description="PostgreSQL max pool size. The maximum number of connections to the PostgreSQL server.",
```
Comment (lferran) [inline nucliadb/nucliadb/ingest/settings.py:68]: copy paste 😛

File: nucliadb/nucliadb/common/maindb/tikv.py [hunk 2]
```diff
@@ -244,32 +353,37 @@ async def reinitialize(self) -> None:
...
                 async with self.connect_lock:
                     logger.warning("Reconnecting to TiKV")
-                    self.tikv = await TransactionClient.connect(self.url)
+                    await self.initialize()
             finally:
                 self.connect_in_progress = False
 
+
+class TiKVDriver(Driver):
+    tikv = None
```
Comment (lferran) [inline nucliadb/nucliadb/common/maindb/tikv.py:362]: not used anymore. can be removed

File: nucliadb/nucliadb/common/maindb/tikv.py [hunk 3]
```diff
@@ -218,21 +213,135 @@ async def count(self, match: str) -> int:
...
+
+class ReadOnlyTiKVTransaction(Transaction):
+    driver: TiKVDriver
+
+    def __init__(self, connection: asynchronous.Snapshot, driver: TiKVDriver):
+        self.connection = connection
+        self.data_layer = TiKVDataLayer(connection)
+        self.driver = driver
+
+    async def abort(self):
```
Comment (lferran) [inline nucliadb/nucliadb/common/maindb/tikv.py:272]: no need to implement abort?

File: nucliadb/nucliadb/common/maindb/tikv.py [hunk 4]
```diff
@@ -218,21 +213,135 @@ async def count(self, match: str) -> int:
...
-                self.tikv = await TransactionClient.connect(self.url)
-            self.initialized = True
+    async def initialize(self) -> None:
+        self._txn_connection = await asynchronous.TransactionClient.connect(self.url)
+
+    async def get_snapshot(
+        self, timestamp: Optional[float] = None, retried: bool = False
+    ) -> asynchronous.Snapshot:
+        if self.connect_in_progress:  # pragma: no cover
+            async with self.connect_lock:
```
Comment (lferran) [inline nucliadb/nucliadb/common/maindb/tikv.py:315]: are both `connect_in_progress` flag and the lock needed? Or could we use just the lock?

File: nucliadb/nucliadb/common/maindb/tikv.py (hunk 3 above)
Comment (vangheem) [reply nucliadb/nucliadb/common/maindb/tikv.py:272]: Nope, snapshot does not have an abort method.

**Review by lferran — APPROVED** [review]

**Pull request merged**